File: api/market.py
import json
import yfinance as yf

def handler(request):
    try:
        ticker = yf.Ticker("^KSE")
        hist = ticker.history(period="30d", interval="1d")
        
        data = []
        for date, row in hist.iterrows():
            data.append({
                "date": date.strftime("%b %d"),
                "close": round(float(row["Close"]), 2),
                "volume": int(row["Volume"]),
                "change": round(float(row["Close"] - row["Open"]), 2)
            })
        
        current = data[-1] if data else {}
        prev = data[-2] if len(data) > 1 else {}
        
        return {
            "statusCode": 200,
            "headers": {"Content-Type": "application/json", "Access-Control-Allow-Origin": "*"},
            "body": json.dumps({
                "history": data,
                "current_price": current.get("close", 0),
                "daily_change_pct": round(
                    ((current.get("close", 0) - prev.get("close", 1)) / prev.get("close", 1)) * 100, 2
                )
            })
        }
    except Exception as e:
        return {"statusCode": 500, "body": json.dumps({"error": str(e)})}

File: api/market.js
export default async function handler(req, res) {
    res.setHeader('Access-Control-Allow-Origin', '*');
    try {
      const response = await fetch(
        'https://query1.finance.yahoo.com/v8/finance/chart/%5EKSE?interval=1d&range=30d'
      );
      const data = await response.json();
      const timestamps = data.chart.result[0].timestamp;
      const closes = data.chart.result[0].indicators.quote[0].close;
  
      const history = timestamps.map((ts, i) => ({
        date: new Date(ts * 1000).toLocaleDateString('en-US', { month: 'short', day: 'numeric' }),
        close: Math.round(closes[i] * 100) / 100
      })).filter(d => d.close);
  
      const current = history[history.length - 1];
      const prev = history[history.length - 2];
      const daily_change_pct = prev
        ? Math.round(((current.close - prev.close) / prev.close) * 10000) / 100
        : 0;
  
      res.json({ history, current_price: current.close, daily_change_pct });
    } catch (e) {
      res.status(500).json({ error: e.message });
    }
  }